### PR TITLE
Add type hinting

### PR DESCRIPTION
**Pull request by greg0ire**
## Changelog

```markdown
### Added
- type hinting on most function signatures
```

## Subject
I avoided touching 2 classes that are most likely be tested because they extend `\TwigTemplate`, and twig isn't even required at the moment, see #158

### CODE REVIEW

File: Adapter/VarnishCache.php [hunk 1]
```diff
@@ -206,7 +206,7 @@ protected function runCommand($command, $expression)
      *
      * @return string
      */
-    protected function getUrl(array $keys)
+    protected function getUrl(array $keys): string
```
Comment (jordisala1991) [inline Adapter/VarnishCache.php]: ?string like the others

Comment (greg0ire) [reply Adapter/VarnishCache.php]: @jordisala1991 fixed

**Comment by greg0ire** [comment]
ping @sonata-project/contributors

**Comment by greg0ire** [comment]
Y U no merge @sonata-project/contributors ?

**Pull request merged**